pkg/detectors/auth0managementapitoken/auth0managementapitoken.go

```diff
@@ -18,35 +19,38 @@ type Scanner struct {
 
 // Ensure the Scanner satisfies the interface at compile time.
 var _ detectors.Detector = (*Scanner)(nil)
+var _ detectors.MaxSecretSizeProvider = (*Scanner)(nil)
 
 var (
 	client = common.SaneHttpClient()
 
 	// long jwt token but note this is default 8640000 seconds = 24 hours but could be set to maximum 2592000 seconds = 720 hours = 30 days
 	// at https://manage.auth0.com/dashboard/us/dev-63memjo3/apis/management/explorer
-	managementApiTokenPat = regexp.MustCompile(detectors.PrefixRegex([]string{"auth0"}) + `\b(ey[a-zA-Z0-9._-]+)\b`)
+	managementAPITokenPat = regexp.MustCompile(`\b(ey[a-zA-Z0-9._-]+)\b`)
 	domainPat             = regexp.MustCompile(`([a-zA-Z0-9\-]{2,16}\.[a-zA-Z0-9_-]{2,3}\.auth0\.com)`) // could be part of url
 )
 
 // Keywords are used for efficiently pre-filtering chunks.
 // Use identifiers in the secret preferably, or the provider name.
-func (s Scanner) Keywords() []string {
-	return []string{"auth0"}
-}
+func (s Scanner) Keywords() []string { return []string{"auth0"} }
+
+const maxSecretSize = 5000
+
+func (Scanner) MaxSecretSize() int64 { return maxSecretSize }
```

Comment:
`MaxSecretSize()` is not used I believe it was intended to be used in `if len(managementAPITokenRes) < 2000 || len(managementAPITokenRes) > 5000 {`

Reply:
We should use it [here](https://github.com/trufflesecurity/trufflehog/blob/a150103e2bc7917baa00bc27af2c15721b649ff0/pkg/engine/ahocorasick/ahocorasickcore.go#L82) to determine the endOffset for the chunk. This ensures we extract only the section of the chunk containing the credential before passing the data to the detector. Consequently, the regex runs on a smaller dataset.